docs/reference/scopes.md

```diff
@@ -0,0 +1,110 @@
+
+This scope grants access to the full Matrix client-server API.
+
+The default policy allows any client and any user to request this scope.
+
+### `urn:matrix:org.matrix.msc2967.client:device:[device id]`
+
+This scope sets the device ID of the session, where `[device id]` is the device ID of the session.
+Currently, MAS only allows the following characters in the device ID: `a-z`, `A-Z`, `0-9` and `-`.
+It also needs to be at least 10 characters long.
+
+There can only be one device ID in the scope list of a session.
+
+The default policy allows any client and any user to request this scope.
+
+### `urn:matrix:org.matrix.msc2967.client:guest`
+
+This scope grants access to a restricted set of endpoints that are available to guest users.
+It is mutually exclusive with the `urn:matrix:org.matrix.msc2967.client:api:*` scope.
+
+Note that MAS doesn't yet implement any special semantic around guest users, but this scope is reserved for future use.
+
+The default policy allows any client and any user to request this scope.
+
+## Synapse-specific scopes
+
+MAS also supports one Synapse-specific scope, which aren't formally defined in any specification.
+
+### `urn:synapse:admin:*`
+
+This scope grants access to the [Synapse admin API].
+
+Because of how Synapse works for now, this scope by itself isn't sufficient to access the admin API.
+A session wanting to access the admin API also needs to have the `urn:matrix:org.matrix.msc2967.client:api:*` scope.
+
+The default policy doesn't allow everyone to request this scope.
+It allows:
+
+- users with the `can_request_admin` attribute set to `true` in the database
+- users listed in the [`policy.data.admin_users`](../reference/configuration.md#policy) configuration option
+
+## MAS-specific scopes
+
+MAS also has a few scopes that are specific to the MAS implementation.
+
+### `urn:mas:graphql:*`
+
+This scope grants access to the whole MAS [GraphQL API].
+What permission the session has on the API is determined by the entity that the session is authorized as.
+If authorized as a user, this will usually allow querying and mutating the user's data.
+
+The default policy allows any client and any user to request this scope.
+
+### `urn:mas:admin`
+
+This scope allows full access to the MAS [GraphQL API].
+It requires the `urn:mas:graphql:*` scope to be present in the request.
+This allows the authenticated entity to perform any operation on the API, regardless of whether the entity owns the data or not.
+
+The default policy doesn't allow everyone to request this scope.
+It allows:
+
+- for the "[authorization code]" and "[device authorization]" grants:
+  - users with the `can_request_admin` attribute set to `true` in the database
+  - users listed in the [`policy.data.admin_users`](../reference/configuration.md#policy) configuration option
+- for the "client credentials" grant:
+  - clients that are listed in the [`policy.data.admin_clients`](../reference/configuration.md#policy) configuration option
```

Comment:
`admin_clients` are not documented there yet only `admin_users`. are clients configured in the configuration file automatically added to the `admin_clients` list?

Reply:
You're right! I forgot to document this, I added this in 31a8add